lib/ansible/modules/cloud/openstack/os_server_action.py

```diff
@@ -239,6 +247,39 @@ def main():
                 _wait(timeout, cloud, server, action, module, sdk)
             module.exit_json(changed=True)
 
+        elif action == 'resize':
+            if server.status not in ['ACTIVE', 'SHUTOFF']:
+                module.exit_json(changed=False, debug='You cannot resize servers in %s' % server.status)
+
+            flavor = cloud.get_flavor(flavor)
+            if flavor is None:
+                module.fail_json(msg="Flavor does not exist")
+            if flavor.id == server.flavor.id:
+                module.exit_json(changed=False, debug='Same flavor as the server')
+
+            cloud.compute.post(
+                _action_url(server.id),
+                json={'resize': {'flavorRef': flavor.id}})
+            # Waiting is not conditional as we need to do a second task
+            for count in sdk.utils.iterate_timeout(
```

Comment:
This is a hanging object, if its not going to be used lets set it to `_`.

``` python
 for _ in sdk.utils.iterate_timeout(
    timeout,
    "Timeout waiting for server to complete %s" % action):
```

Reply:
Sorry for the delay, I've just commited a change that fixes that nit.